tools/data_gen/src/main.rs

```diff
@@ -0,0 +1,173 @@
+#[macro_use]
+extern crate lazy_static;
+
+use argh::FromArgs;
+use calyx::{errors::CalyxResult, frontend, ir};
+use serde::ser::{Serialize, Serializer};
+use serde_json::{json, Map, Value};
+use std::borrow::Borrow;
+use std::cell::{Ref, RefCell};
+use std::collections::HashMap;
+use std::path::{Path, PathBuf};
+use std::rc::Rc;
+
+lazy_static! {
+    static ref SIZEMAP: HashMap<&'static str, Vec<&'static str>> = {
+        let mut m = HashMap::new();
+        m.insert("std_mem_d1", vec!["SIZE"]);
+        m.insert("std_mem_d2", vec!["D0_SIZE", "D1_SIZE"]);
+        m.insert("std_mem_d3", vec!["D0_SIZE", "D1_SIZE", "D2_SIZE"]);
+        m.insert(
+            "std_mem_d4",
+            vec!["D0_SIZE", "D1_SIZE", "D2_SIZE", "D3_SIZE"],
+        );
+        m
+    };
+}
+
+struct CellData {
+    name: String,
+    width: u64,
+    sizes: Vec<u64>,
+}
+
+enum SizeVec {
+    Quad(Vec<Vec<Vec<Vec<u64>>>>),
+    Triple(Vec<Vec<Vec<u64>>>),
+    Double(Vec<Vec<u64>>),
+    Single(Vec<u64>),
+}
+
+impl Serialize for SizeVec {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match &*self {
+            SizeVec::Single(v) => {
+                serializer.serialize_newtype_variant("SizeVec", 3, "data", &v)
+            }
+            SizeVec::Double(v) => {
+                serializer.serialize_newtype_variant("SizeVec", 2, "data", &v)
+            }
+            SizeVec::Triple(v) => {
+                serializer.serialize_newtype_variant("SizeVec", 1, "data", &v)
+            }
+            SizeVec::Quad(v) => {
+                serializer.serialize_newtype_variant("SizeVec", 0, "data", &v)
+            }
+        }
+    }
+}
+
+#[derive(FromArgs)]
+/// Used to read a file
+struct FilePath {
+    /// file path to read data from
+    #[argh(positional, from_str_fn(read_path))]
+    file: Option<PathBuf>,
+}
+
+fn read_path(path: &str) -> Result<PathBuf, String> {
+    Ok(Path::new(path).into())
+}
+
+fn main() -> CalyxResult<()> {
+    let fp: FilePath = argh::from_env();
+
+    let lib_path: &Path = &Path::new("../../../calyx");
```

Comment:
Let's make this an argument to the tool in the same way it is an argument in the Calyx compiler. The default value can be set to the same thing as the Calyx compiler. Setting it to `../../../calyx` might not work correctly from different directories.